### ISSUE TITLE
Refactor LrSchedulerHook.__init__(by_epoch,warmup,warmup_iters,warmup_ratio,kwargs)

### ISSUE THREAD

**Issue by amirassov** [ppln-team/ppln#13]
I've selected [**LrSchedulerHook.__init__(by_epoch,warmup,warmup_iters,warmup_ratio,kwargs)**](https://github.com/amirassov/ppln/blob/7eb72da7571dbbd312ab7274d420b723658245b2/ppln/hooks/lr_scheduler.py#L9-L29) for refactoring, which is a unit of **18** lines of code and **5** parameters. Addressing this will make our codebase more maintainable and improve [Better Code Hub](https://bettercodehub.com)'s **Keep Unit Interfaces Small** guideline rating! 👍 

Here's the gist of this guideline:
- **Definition** 📖 
Limit the number of parameters per unit to at most 4.
- **Why**❓ 
Keeping the number of parameters low makes units easier to understand, test and reuse.
- **How** 🔧 
Reduce the number of parameters by grouping related parameters into objects. Alternatively, try extracting parts of units that require fewer parameters.

You can find more info about this guideline in [Building Maintainable Software](http://shop.oreilly.com/product/0636920049159.do). 📖 
 
---- 
ℹ️ To know how many _other_ refactoring candidates need addressing to get a guideline compliant, select some by clicking on the 🔲  next to them. The risk profile below the candidates signals (✅) when it's enough! 🏁 

----
Good luck and happy coding! :shipit: :sparkles: :100:
